packages/pyright-internal/src/analyzer/docStringUtils.ts

```diff
@@ -55,53 +55,133 @@ export function cleanAndSplitDocString(rawString: string): string[] {
+// ```python
+// def func(param1, param2):
+//     """
+//     Description of function (we don't parse this)
+//
+//     Args:
+//         param1: multi-line description first line
+//             multi-line description second line
+//
+//     Returns:
+//         Optional description of return type (we don't parse this)
+//
+//     Raises:
+//         Optional description of potential errors raised (we don't parse this)
+//     """
+//     pass
+// ```
+//
+// Google, variant 2
+// ```python
+// def func(param1, param2):
+//     """
+//     Description of function (we don't parse this)
+//
+//     Args:
+//         param1 (type): multi-line description first line
+//             multi-line description second line
+//
+//     Returns:
+//         Optional description of return type (we don't parse this)
+//
+//     Raises:
+//         Optional description of potential errors raised (we don't parse this)
+//     """
+//     pass
+// ```
 export function extractParameterDocumentation(functionDocString: string, paramName: string): string | undefined {
     if (!functionDocString || !paramName) {
         return undefined;
     }
 
-    // Python doesn't have a single standard for param documentation. There are three
-    // popular styles.
-    //
-    // 1. Epytext:
-    //      @param param1: description
-    // 2. reST:
-    //      :param param1: description
-    // 3. Google (variant 1):
-    //      Args:
-    //          param1: description
-    // 4. Google (variant 2):
-    //      Args:
-    //          param1 (type): description
-
     const docStringLines = cleanAndSplitDocString(functionDocString);
-    for (const line of docStringLines) {
-        const trimmedLine = line.trim();
-
-        // Check for Epytext
-        let paramOffset = trimmedLine.indexOf('@param ' + paramName);
-        if (paramOffset >= 0) {
-            return trimmedLine.substr(paramOffset + 7);
-        }
-
-        // Check for reST format
-        paramOffset = trimmedLine.indexOf(':param ' + paramName);
-        if (paramOffset >= 0) {
-            return trimmedLine.substr(paramOffset + 7);
+    const docstringTypes = [
+        { marker: `@param ${paramName}`, trim: 7 },
+        { marker: `:param ${paramName}:`, trim: 7 },
+        { marker: `${paramName}:`, trim: 0 },
+        { marker: `${paramName} (`, trim: 0 },
+    ];
+
+    // Iterate over each line and check if it matches one of the above markers.
+    for (const [idx, line] of docStringLines.entries()) {
+        let ret = undefined;
+        let paramOffset = -1;
+
+        for (const typ of docstringTypes) {
+            // If we get a match, process the remaining lines of the docstring to check for multi-line parameter descriptions.
+            paramOffset = line.indexOf(typ.marker);
+            if (paramOffset >= 0) {
+                ret = parseParameterDescription(paramOffset, typ.trim, docStringLines.slice(idx));
+                if (ret) {
+                    return ret;
```

Comment:
nit: i prefer to avoid abbreviating variable names if possible just because it makes code more readable imo
```suggestion
        let result = undefined;
        let paramOffset = -1;

        for (const typ of docstringTypes) {
            // If we get a match, process the remaining lines of the docstring to check for multi-line parameter descriptions.
            paramOffset = line.indexOf(typ.marker);
            if (paramOffset >= 0) {
                result = parseParameterDescription(paramOffset, typ.trim, docStringLines.slice(idx));
                if (result) {
                    return result;
```